src/skmultiflow/lazy/base_neighbors.py

```diff
@@ -0,0 +1,41 @@
+from skmultiflow.core import BaseSKMObject
+from skmultiflow.utils import SlidingWindow
+
+from sklearn.neighbors import KDTree
+
+
+class BaseNeighbors(BaseSKMObject):
+    """ Base class for neighbors-based estimators
+    """
+    def __init__(self,
+                 n_neighbors=5,
+                 max_window_size=1000,
+                 leaf_size=30,
+                 metric='euclidean'):
+        self.n_neighbors = n_neighbors
+        self.max_window_size = max_window_size
+        self.leaf_size = leaf_size
+        if metric not in self.valid_metrics():
+            raise ValueError("Invalid metric: {}.\n"
+                             "Valid options are: {}".format(metric,
+                                                            self.valid_metrics()))
+        self.metric = metric
+        self.data_window = SlidingWindow(window_size=max_window_size)
+
+    def _get_neighbors(self, X):
+        tree = KDTree(self.data_window.features_buffer, self.leaf_size, metric=self.metric, )
```

Comment:
Is there a reason for the comma at the end of KDTree's construction?

Reply:
No. Removed in 31687ca4743f3fa03f752ed182135cbef9fc7901